addons/binding/org.openhab.binding.airquality/src/main/java/org/openhab/binding/airquality/handler/AirQualityHandler.java

```diff
@@ -0,0 +1,308 @@
+
+            // Map the JSON response to an object
+            result = gson.fromJson(repsonse, AirQualityJsonResponse.class);
+            if (result == null) {
+                errorMsg = "no data returned";
+            } else if (result.getStatus() == "error") {
+                errorMsg = result.getData().toString();
+            } else if (result.getData() != null) {
+                resultOk = true;
+            } else {
+                errorMsg = "missing data sub-object";
+            }
+
+            if (!resultOk) {
+                logger.warn("Error in aqicn.org (Air Quality) response: {}", errorMsg);
+            }
+
+        } catch (MalformedURLException e) {
+            errorMsg = e.getMessage();
+            logger.warn("Constructed url is not valid: {}", errorMsg);
+        } catch (IOException e) {
+            errorMsg = e.getMessage();
+            logger.warn("Error running aqicn.org (Air Quality) request: {}", errorMsg);
+        } catch (JsonSyntaxException e) {
+            errorMsg = e.getMessage();
+            logger.warn("Error parsing aqicn.org (Air Quality) response: {}", errorMsg);
+        }
+
+        // Update the thing status
+        if (resultOk && result != null) {
+            String attributions = result.getData().getAttributions();
+            updateStatus(ThingStatus.ONLINE, ThingStatusDetail.NONE, attributions);
+        } else {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.OFFLINE.COMMUNICATION_ERROR, errorMsg);
+        }
+
+        return resultOk ? result : null;
+    }
+
+    public static Object getValue(String channelId, AirQualityJsonResponse data) throws Exception {
+        String[] fields = StringUtils.split(channelId, "#");
+
+        if (data == null) {
+            return null;
+        }
+
+        String fieldName = fields[0];
+
+        switch (fieldName) {
+            case AirQualityBindingConstants.AQI:
+                return data.getData().getAqi();
+            case AirQualityBindingConstants.AQIDESCRIPTION:
+                return data.getData().getAqiDescription();
+            case AirQualityBindingConstants.PM25:
+                return data.getData().getIaqi().getPm25();
+            case AirQualityBindingConstants.PM10:
+                return data.getData().getIaqi().getPm10();
+            case AirQualityBindingConstants.O3:
+                return data.getData().getIaqi().getO3();
+            case AirQualityBindingConstants.NO2:
+                return data.getData().getIaqi().getNo2();
+            case AirQualityBindingConstants.CO:
+                return data.getData().getIaqi().getCo();
+            case AirQualityBindingConstants.LOCATIONNAME:
+                return data.getData().getCity().getName();
+            case AirQualityBindingConstants.LOCATIONURL:
+                return data.getData().getCity().getUrl();
+            case AirQualityBindingConstants.OBSERVATIONTIME:
+                return data.getData().getTime().getS();
```

Comment:
Specifically for the purpose that @kaikreuzer explicitly asked for reviewers' approval, I've started a new review with required changes. The timezone bug should be finished before merging.